### ISSUE TITLE
Show generated recommendations

### ISSUE THREAD

**Issue by fredhersch** [google/android-fhir#626]
**User story**
As a HCW I can generate and view the recommendations for a patient on completion of a screener flow

**Steps:**
- User has completed the screener questionnaire - #625 
- Selects the option to Show Recommendations
- This will trigger a process to determine the triage status and select he CarePlan for the patient (in the future it will generate the CarePlan)
-- Determining Triage Status will result in the creation of a RiskAssessment Resource - see additional information section
- Based on the RiskAssessment a CarePlan resource will be created (from a template) for the specific patient
- Using the Tasks on the CarePlan, the recommendations are shown in the app

**Alt path:**
- If there is an error generating the recommendations, return that to the user

**Additional information:**

Rules for generating the RiskAssessment and selecting the CarePlan are as follows:
```
If Sp02 < 90% THEN
   HIGH_RISK
   Select CarePlan_HighRisk.json
Else If Sp02 >90 - <94% THEN
   MODERATE_RISK
   Select CarePlan_ModRisk.json
Else If Presence_Of_Symptoms THEN
   MODERATE_RISK
   Select CarePlan_ModRisk.json
Else If Sp02 > 94% AND Any Comorbidities THEN
   MODERATE_RISK
   Select CarePlan_ModRisk.json
Else If Sp02 > 94% AND NO CO-morbidities THEN
   LOW_RISK
   Select CarePlan_LowRisk.json
```

**Generating the RiskAssessment Resource**

Use the data-access API to create a new RiskAssessment resource and populate with:

- TBC

**Selecting a CarePlan template**

- A set of json files representing CarePlan templates will be added to the application
- Based on the assessment, the relevant CarePlan_<>.json will be selected
- This will need to be modified accordingly and saved as the CarePlan for the patient

**Comment by jingtang10** [comment]
@santosh-pingle to work on the generation of these risk assessments and @aditya-07 to work on the display of these in the patient card

**Comment by jingtang10** [comment]
@fredhersch can you provide an example care plan while aditya and santosh work on the generation / display of risk assessment. thanks!

**Comment by jingtang10** [comment]
@aditya-07 pleaes also within this ticket address the ui display of high/mid/low risks (red/amber/green dots)

**Comment by jingtang10** [comment]
Regarding the rules @fredhersch posted - ignore the care plan part for now, and use https://www.hl7.org/fhir/riskassessment.html and https://www.hl7.org/fhir/riskassessment-definitions.html#RiskAssessment.prediction.qualitativeRisk with this value set: https://www.hl7.org/fhir/valueset-risk-probability.html

**Comment by santosh-pingle** [comment]
Generating risk-assessment resource.

**Comment by santosh-pingle** [comment]
> @santosh-pingle + @aditya-07 to work on the generation of these risk assessments and @aditya-07 to work on the display of these in the patient card

@jingtang10 @aditya-07 
Can you please explain in more details about implementing display in the patient card?

**Comment by santosh-pingle** [comment]
@aditya-07 
Is below PR blocker to implement additional display changes of patient card, if yes can we please get merged on priority?
https://github.com/google/android-fhir/pull/691

**Comment by santosh-pingle** [comment]
> Generating risk-assessment resource.

I will push PR about generating risk-assessment in first half of the day today.

**Comment by santosh-pingle** [comment]
Generated risk assessment resource
```
{"resourceType":"RiskAssessment","id":"88b73f54-fd85-4fac-8bb7-2bb917de735a","subject":{"reference":"Patient/2126234"},"encounter":{"reference":"Encounter/c5fd5464-75ec-4bb7-b9c0-28e59a94b1c2"},"prediction":[{"qualitativeRisk":{"coding":[{"system":"http://terminology.hl7.org/CodeSystem/risk-probability","code":"moderate","display":"Moderate likelihood"}]}}]}
```

**Comment by santosh-pingle** [comment]
#701 need to fix this issue to unblock ui changes further

**Comment by santosh-pingle** [comment]
Working on review comments of #699

**Comment by santosh-pingle** [comment]
Risk assessment UI Display action items :

- Fetch Risk assessment resource.
- Update color according to risk probability on Patient card.
- Refresh patient card ui after completing screener survey for latest risk probability.
- Show risk probability color on patient search page as well.

**Comment by santosh-pingle** [comment]
In progress : Fetch Risk assessment resource.

**Comment by santosh-pingle** [comment]
@jingtang10 @fredhersch @aditya-07 

What will be initial value of risk assessment **status**?
It is required to show on patient details card screen.

**Comment by santosh-pingle** [comment]
@jingtang10 @fredhersch @aditya-07 
It is also required to show last contacted value on patient details screen, what will be value for it?
is it occurrence date time type when risk assessment resource was generated?

**Comment by aditya-07** [comment]
> @jingtang10 @fredhersch @aditya-07
> 
> What will be initial value of risk assessment **status**?
> It is required to show on patient details card screen. (are we going to show risk probability code as status or risk assessment resource field status ?)

Status is UNKNOWN if we don't have a risk assessment for a Patient.

**Comment by aditya-07** [comment]
> @jingtang10 @fredhersch @aditya-07
> It is also required to show last contacted value on patient details screen, what will be value for it?
> is it occurrence date time type when risk assessment resource was generated?

Yes, I think so.

**Comment by santosh-pingle** [comment]
I will push PR in second half of today which will include
1. patient risk with color code on patient search page.
2. refresh patient risk color code on search page after taking screener encounter.
3. Patient risk color code on patient display card.
4. refresh patient risk color after taking screener encounter 
5. Fetch risk assessment for given patient.
6. fetch risk assessment for all patients on patient search page.

**Comment by santosh-pingle** [comment]
There are changes in parent PR #691, therefore it is required to revise/update current PR changes.
I will be working on it.

**Comment by Tarun-Bhardwaj** [comment]
> Regarding the rules @fredhersch posted - ignore the care plan part for now, and use https://www.hl7.org/fhir/riskassessment.html and https://www.hl7.org/fhir/riskassessment-definitions.html#RiskAssessment.prediction.qualitativeRisk with this value set: https://www.hl7.org/fhir/valueset-risk-probability.html

@fredhersch , generation and display of risk assessment is done and changes merged in master. Based on the comment above, do we drop the care plan for now and have another issue created for that or should the care plan be done as part of this issue?

**Comment by Tarun-Bhardwaj** [comment]
> Regarding the rules @fredhersch posted - ignore the care plan part for now, and use https://www.hl7.org/fhir/riskassessment.html and https://www.hl7.org/fhir/riskassessment-definitions.html#RiskAssessment.prediction.qualitativeRisk with this value set: https://www.hl7.org/fhir/valueset-risk-probability.html

@jingtang10 , @fredhersch , changes have been merged in master for "Risk assessment generation display". Can we close this issue, make it part of Q3 release and create a new one for "Care plan generation" that can be worked on as part of Q4 release? CC: @jtournier23 , @santosh-pingle

**Comment by jingtang10** [comment]
We'll create another issue for care plan. The risk assessment generation is done and dusted.
